core/src/test/java/io/grpc/internal/DelayedStreamTest.java

```diff
@@ -154,11 +155,15 @@ public void setStream_cantCreateTwice() {
 
   @Test
   public void streamCancelled() {
-    stream.cancel(Status.CANCELLED);
+    Status status = Status.CANCELLED.withDescription("unique status message");
+    stream.cancel(status);
+    verify(listener).closed(same(status), isA(Metadata.class));
+    verifyNoMoreInteractions(listener);
 
-    // Should be a no op, and not fail due to transport not returning a newStream
+    // Should be _mostly_ a no op, and not fail due to transport not returning a newStream
     stream.setStream(realStream);
 
-    verify(listener).closed(eq(Status.CANCELLED), isA(Metadata.class));
+    verifyNoMoreInteractions(listener);
```

Comment:
FWIW this is kind of tight coupling between the SUT and the test.  Checking that listener is closed is likely sufficient.